lib/spack/spack/bootstrap.py

```diff
@@ -543,7 +554,16 @@ def ensure_executables_in_path_or_raise(executables, abstract_spec):
         b = _make_bootstrapper(current_config)
         try:
             if b.try_search_path(executables, abstract_spec):
-                return
+                # Additional environment variables needed
+                env_mods = spack.util.environment.EnvironmentModifications()
+                cmd = spack.util.executable.which(*executables)
+                concrete_spec = spack.store.db.query(abstract_spec, installed=True)[0]
+                for dep in concrete_spec.traverse(root=True, order='post'):
```

Comment:
Can we avoid this duplicated load logic? It's not entirely correct as it adds environment modifications for build deps and their dependencies too, as I understand.

Reply:
Good point. I think we can avoid this, but I'd prefer to refactor that in another PR. For what is worth these lines have been extracted from the function that has been deleted and they are needed to set `PYTHONPATH` and similar variables.

I'll remove build deps though as the point makes total sense to me.

Reply:
See b04d20b